src/lib.rs

```diff
@@ -4221,6 +4270,161 @@ fn map_darwin_target_from_rust_to_compiler_architecture(target: &Target) -> &str
     }
 }
 
+// Rust and clang/cc don't agree on what equivalent flags should look like either.
+fn rustc_to_cc_flag(flag: &str, family: ToolFamily) -> Option<OsString> {
+    match family {
+        ToolFamily::Clang { .. } | ToolFamily::Gnu => match flag {
+            _ if flag.starts_with("branch-protection") => {
+                Some(format!("-m{}", flag.replace(",", "+")).into())
+            }
+            _ if flag.starts_with("code-model") => {
```

Comment:
I'd prefer `flag.starts_with("code-model=")` or similar, to ensure that we don't match here if a new flag like `-Ccode-model-` was added.

Actually, I'd rather prefer that the code did `flag.split_once('=')` all the way at the start.

Reply:
> I'd prefer flag.starts_with("code-model=") or similar, to ensure that we don't match here if a new flag like -Ccode-model- was added.

Is that really a concern we have? If so we can match with the '=' for sure, but the example you have would go against how all the rustc flags currently work so I'd be pretty surprised if someone added it.

> Actually, I'd rather prefer that the code did flag.split_once('=') all the way at the start.

Well we can't do that because some of the codegen flags don't contain a '='. Either they just don't take it or the lack of it implies a certain option.

Reply:
Hm that's true, yeah I can make the matching more strict, sure :)